### ISSUE TITLE
Using a Symbol() in extension API is broken in insiders (works in Stable)

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#57513]
In order to expose some API from my extension to the tests without it being part of public API, I used a `Symbol`:

symbols.ts
```ts
export const internalApiSymbol = Symbol();
```

extension.ts
```ts
return {
		[internalApiSymbol]: {
			analyzerCapabilities: analyzer.capabilities,
			currentAnalysis: () => analyzer.currentAnalysis,
			daemonCapabilities: flutterDaemon ? flutterDaemon.capabilities : DaemonCapabilities.empty,
			debugProvider,
			initialAnalysis,
			nextAnalysis,
			reanalyze,
			renameProvider,
			sdks,
			testTreeProvider,
		},
	};
```

Then in my tests I accessed it like this:

```ts
export async function activateWithoutAnalysis(): Promise<void> {
	log("Activating");
	await ext.activate();
//	if (!ext.isActive)
//		throw new Error("Extension is not active after calling activate()");
//	console.log(inspect(ext.exports));
	extApi = ext.exports[internalApiSymbol];
	if (!extApi)
		throw new Error("Extension API not found. Did the extension activate correctly?");
}
```

In Insiders, this throws on the last line. `extApi` always comes back undefined. In Stable, I get back my API as expected. I've verified that the last line of my activate method is being run correctly. I'm not sure what's change might have caused this.

**Comment by jrieken** [comment]
Works just fine for me

**Comment by DanTup** [comment]
Ugh, let me try and get a repro running on AppVeyor or something. Out of interest, what platform did you test on? Both places I saw it were Windows - it looked like it might not have happened on macOS/Linux (I figured I must've been mistaken since that seems weird, but I guess it's possible).

**Comment by DanTup** [comment]
I can repro using the Hello World extension with [a simple test](https://github.com/DanTup/ext-symbol-err/commit/d8f6ddc6fc02f35b27db53c89556874bedcb44d1):

https://github.com/DanTup/ext-symbol-err

![ext err](https://user-images.githubusercontent.com/1078012/44797118-52fc7e00-aba6-11e8-8c69-c11911c54b3d.png)

If you can't repro (on Windows), let me know if it's worth me setting up on AppVeyor or something.

**Comment by jrieken** [comment]
If this isn't a fluke then it's the electron update. I have tested on macOS with a simple Symbol and using that in an export.

**Comment by bpasero** [comment]
@jrieken 

> If this isn't a fluke then it's the electron update.

There was no electron update this month?

**Comment by jrieken** [comment]
@bpasero really? I see 1.26 using 2.0.5 and 1.27 using 2.0.7

**Comment by DanTup** [comment]
Also tested on latest Insiders on macOS and it passed, so certainly does seem Windows-specific.

**Comment by bpasero** [comment]
@jrieken yeah, that not an update that I would think can cause a regression like this, I would put my money on the `stdfork` change rather. 

@DanTup are you able to run VSCode out of sources on Windows? That would allow us to rule out the Electron 2.0.7 update as cause for this.

**Comment by DanTup** [comment]
It's not something I've done/tried before, so that might depend on how complicated it is and/or the dependencies. I'll have a look shortly.

**Comment by DanTup** [comment]
Ok, I think I'm up and running:

![codecode](https://user-images.githubusercontent.com/1078012/44803134-7aa71280-abb5-11e8-8c49-80467eae301d.png)

I'll need some instructions on what you want me to try though!

**Comment by DanTup** [comment]
Not sure what I'm doing wrong, but now when I try to run it I see:

```
[18:37:57] [marketplace] ms-vscode.node-debug@1.27.6 ✔︎
[18:37:57] [marketplace] ms-vscode.node-debug2@1.27.1 ✔︎

{ errorCode: 'load',
  moduleId: 'vs/code/electron-main/main',
  neededBy: [ '===anonymous1===' ],
  detail:
   { Error: ENOENT: no such file or directory, open 'M:\Coding\Applications\vscode\out\vs\code\electron-main\main.js'
     errno: -4058,
     code: 'ENOENT',
     syscall: 'open',
     path: 'M:\\Coding\\Applications\\vscode\\out\\vs\\code\\electron-main\\main.js' } }
```

I've run `yarn run watch`, pressed Ctrl+Shift+B and then run code.bat :/

**Comment by bpasero** [comment]
@DanTup to start fresh, try with `git clean -xfd` followed by `yarn` and then `yarn watch`

**Comment by DanTup** [comment]
Tried that, and have the same issue:

```
[12:27:06] Downloading extension: ms-vscode.node-debug2@1.27.1 ...
[12:27:06] Downloading extension: ms-vscode.node-debug@1.27.6 ...
[12:27:09] [marketplace] ms-vscode.node-debug ✔︎
[12:27:09] [marketplace] ms-vscode.node-debug2 ✔︎

{ errorCode: 'load',
  moduleId: 'vs/code/electron-main/main',
  neededBy: [ '===anonymous1===' ],
  detail:
   { Error: ENOENT: no such file or directory, open 'M:\Coding\Applications\vscode\out\vs\code\electron-main\main.js'
     errno: -4058,
     code: 'ENOENT',
     syscall: 'open',
     path: 'M:\\Coding\\Applications\\vscode\\out\\vs\\code\\electron-main\\main.js' } }
```

**Comment by DanTup** [comment]
(However, that file actually exists - I can Ctrl+Click it and it opens)

**Comment by DanTup** [comment]
Scratch that - restarting the task worked. I think the compile was still happening (it stalled for a long time, but now I see it wrote `Finished compilation` in the meantime)...

Still fails on that version:

![fail](https://user-images.githubusercontent.com/1078012/45154786-aa1ad800-b1d0-11e8-9d8e-cb7ad4336d2e.png)

**Comment by DanTup** [comment]
I think this may be related to https://github.com/Microsoft/vscode/issues/58815 - I think something on Windows may be split across processes that previously was not?

**Comment by DanTup** [comment]
So this turns out to be related to https://github.com/Microsoft/vscode/issues/58388. Forcing a lowercase filename for loading the test seems to resolve the issue 🤷‍♂️

**Comment by DanTup** [comment]
(This was fixed with #58388 :-))
